### ISSUE TITLE
Training Stability

### ISSUE THREAD

**Issue by alecGraves** [alecGraves/WolfDetector#2]
There appears to be a lot of training stability issues with the computational graph used for the YOLOv2 loss function.

See [this post](https://community.wolfram.com/groups/-/m/t/2295262) for more details.

I have gotten around a lot of them by using smoother loss functions and thresholding operations not present in the original YOLOv2 implementation.

It would be great to understand why training is so unstable and move to a more standard loss function.

Hopefully, this is just something that will be fixed as MXNet and Wolfram neural network functionality improve.

**Comment by alecGraves** [comment]
Even with the fixes applied, there can still be a "parameters of the net diverged" error. I would estimate around one every 20 runs fails to train on the fruits dataset.
